addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -31,15 +31,13 @@ def _find_unreservable_moves(self, batch_size=1000, return_fast=False):
         # This method is patterned on SaleOrder._find_unfulfillable_order_lines
         # in udes_sale_stock.  Changes applied there should be considered for
         # this method too.
-        Location = self.env["stock.location"]
         Move = self.env["stock.move"]
         Quant = self.env["stock.quant"]
 
         # Create empty record sets for moves.
         unreservable_moves = Move.browse()
 
-        # Get unreserved stock for each product in locations
-        locations = Location.get_available_stock_locations()
+        location = self.picking_type_id.default_location_src_id
```

Comment:
Why not `picking.location_id` after line 66 instead of here? Because picking type `default_location_src_id` could be technically different than the one of the picking.

Reply:
What if a picking had its location manually changed but not the others? The product counts could be misleading at that point.

Reply:
i.e you have Pick to pick from PICK ZONE

PICK1 - ZONE A
PICK2 - ZONE B
PICK3 - PICK ZONE
PICK4 - PICK ZONE

So probably would be safest to use the picking type location?